projenrc/github-workflows.ts

```diff
@@ -493,3 +493,163 @@ export function runCommitLintWorkflow(project: AwsCdkConstructLibrary) {
+    permissions: {
+      idToken: JobPermission.WRITE, //needed to interact with GitHub's OIDC Token endpoint.
+      contents: JobPermission.READ,
+    },
+    outputs: {
+      [PATCH_CREATED_OUTPUT]: {
+        stepId: CREATE_PATCH_STEP_ID,
+        outputName: PATCH_CREATED_OUTPUT,
+      },
+    },
+    steps: [
+      {
+        name: 'Checkout project',
+        uses: 'actions/checkout@v3',
+        with: {
+          ref: 'main',
+        },
+      },
+      {
+        name: 'Setup Node.js',
+        uses: 'actions/setup-node@v3',
+        with: {
+          'node-version': '20.x',
+        },
+      },
+      {
+        name: 'Install dependencies',
+        run: 'yarn install --check-files --frozen-lockfile',
+      },
+      {
+        name: 'Setup AWS credentials',
+        uses: 'aws-actions/configure-aws-credentials@v3.0.1',
+        with: {
+          'role-to-assume': '${{ secrets.AWS_ROLE_ARN_TO_ASSUME }}',
+          'aws-region': '${{ env.AWS_REGION }}',
+          'role-duration-seconds': '7200',
+        },
+      },
+      {
+        name: 'Run code generation',
+        run: 'npx projen generate-models-containers',
+      },
+      {
+        name: 'Find mutations',
+        id: CREATE_PATCH_STEP_ID,
+        run: [
+          'git add .',
+          'git diff --staged --patch --exit-code > .repo.patch || echo "patch_created=true" >> $GITHUB_OUTPUT',
+        ].join('\n'),
+      },
+      {
+        name: 'Upload patch',
+        if: `steps.${CREATE_PATCH_STEP_ID}.outputs.${PATCH_CREATED_OUTPUT}`,
+        uses: 'actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32',
+        with: {
+          name: '.repo.patch',
+          path: '.repo.patch',
+        },
+      },
+    ],
+  };
+
+  // second job to create a PR if any changes were detected
+  const pr: Job = {
+    name: 'Code generation',
+    needs: ['generate'],
+    runsOn: ['ubuntu-latest'],
+    permissions: {
+      contents: JobPermission.READ,
+    },
+    if: '${{ needs.upgrade.outputs.patch_created }}',
+    steps: [
+      {
+        name: 'Checkout',
+        uses: 'actions/checkout@v3',
+        with: {
+          ref: 'main',
+        },
+      },
+      {
+        name: 'Download patch',
+        uses: 'actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a',
+        with: {
+          name: '.repo.patch',
+          path: '${{ runner.temp }}',
+        },
+      },
+      {
+        name: 'Apply patch',
+        run: '[ -s ${{ runner.temp }}/.repo.patch ] && git apply ${{ runner.temp }}/.repo.patch || echo "Empty patch. Skipping."',
+      },
+      {
+        name: 'Set git identity',
+        run: [
+          'git config user.name "github-actions"',
+          'git config user.email "github-actions@github.com"',
+        ].join('\n'),
+      },
+      {
+        name: 'Create Pull Request',
+        id: 'create-pr',
+        uses: 'peter-evans/create-pull-request@38e0b6e68b4c852a5500a94740f0e535e0d7ba54',
```

Comment:
`v4` or `v5`:
```suggestion
        uses: 'peter-evans/create-pull-request@v4',
```

Reply:
done in [082cc46](https://github.com/awslabs/generative-ai-cdk-constructs/pull/238/commits/082cc46cc0456acc6557d153b746b9921785ed63)